### PR TITLE
Cldr to locales

### PR DESCRIPTION
**Pull request by ylavoie**
Remove Locale::CLDR in favor of Locales to significantly lower the amount of dependencies.

### CODE REVIEW

**Comment by ylavoie** [comment]
Locales 0,34 is using CLDR data from 2016 and misses some variants like ar_EG, es_MX and a few others. Locale::CLDR was using 2014 data, so we've improved but bringing data up to CLDR v41 would be desirable.

File: lib/LedgerSMB/I18N.pm [hunk 1]
```diff
@@ -127,11 +126,12 @@ Get a country localized list to allow user selection
...
 sub get_country_list {
     my $language = shift;
-    my %regions = Locale::CLDR->new($language)->all_regions->%*;
+    my $locale = Locales->new($language);
+    my %regions = $locale->get_territory_lookup();
     return [
         sort { $a->{text} cmp $b->{text} }
         map { +{ value => uc($_),
-                 text  => $regions{$_} }
+                 text  => $regions{lc($_)} }
```
Comment (ehuelsmann) [inline lib/LedgerSMB/I18N.pm:134]: When iterating over the keys of `%regions`, I'd expect you shouldn't be transforming the key before doing a lookup (i.e. no `lc()`)?

**Review by ehuelsmann — COMMENTED** [review]
Should we wait for `Locales` to be updated? From what I understood, the version it uses was 29 with 40 the latest release and there's an outstanding request to the author to update it?

File: lib/LedgerSMB/I18N.pm (hunk 1 above)
Comment (ylavoie) [reply lib/LedgerSMB/I18N.pm]: The keys are already lowercased but I fixed it.

**Comment by ylavoie** [comment]
> Should we wait for `Locales` to be updated? From what I understood, the version it uses was 29 with 40 the latest release and there's an outstanding request to the author to update it?

I'm pushing on the maintainer to update `Locales` to a more recent `CLDR` version. I'd say let's wait a few weeks, to give him time to answer

**Pull request closed without merging**